### ISSUE TITLE
Network storage: SMB Password Issue

### ISSUE THREAD

**Issue by wtaferner** [home-assistant/supervisor#4953]
### The problem

It seems that certain characters within a password do flaw a successful authentication.

Not 100% sure if it an issue of HA, but I think it is, but give it a shot yourself as I post the failing password here.

`~L_{rdbH."@\0w)Gt^o6`

**How to reproduce**
- Create a Samba User on a remote storage with the password above and a proper share
- Try to add this share with the user created in HA

CIFS: Attempting to mount \\192.168.xxx.xxx\share_name
CIFS: Status code returned 0xc000006d STATUS_LOGON_FAILURE
CIFS: VFS: \\192.168.xxx.xxx Send error in SessSetup = -13
CIFS: VFS: cifs_mount failed w/return code = -13

### What version of Home Assistant Core has the issue?

core-2024.3.0

### What was the last working version of Home Assistant Core?

core-2024.3.0

### What type of installation are you running?

Home Assistant OS

**Comment by agners** [comment]
Hm, since PR #4399 Supervisor uses a credentials file, with which any password should be working :thinking: 

Are you sure this is on Supervisor side and not server side? As in, does accessing with that username/password from another system works?

**Comment by wtaferner** [comment]
@agners 
I am not 100% sure that it is a HA issue, this is why I posted the password to provide a possibility to test in any other setting as well. My guess is some escaping/replacing issue along the way. 

For my own setting, it is an Unraid Server where I use their GUI to enter the password and HA where I use the GUI to enter the password in the Network Storage Dialog to enter the password, so yeah, the issue can be anywhere between the two. 

So, if you can use this password with you SMB set-up, I guess we can close this issue...then it seems to be related to Unraid not properly saving the password via their GUI and therefore not matching.

**Comment by agners** [comment]
Ok, I've tried to reproduce this on my end. I've used another Home Assistant installation with the Samba add-on as SMB server side. And I've ended up uncovering a bug in the handling of that password in there :see_no_evil:  (fixed with https://github.com/home-assistant/addons/pull/3508).

Once I am using that fixed version, I can use that password to authenticate from my host PC.

I then moved to use that password using Supervisor's Network storage feature. And the password worked here too.

So it seems that the Unraid Server is at fault in your case. I am closing this as there seems to be no issue from Supervisor's perspective.
